frontend/src/service/apiService.ts

```diff
@@ -1,39 +1,41 @@
 import {Game, LoginData, LoginResponse, MyUserCreationData, UserGameDTO} from "./model";
 import axios, {AxiosResponse} from "axios";
 
-let requestConfig = {
-    headers: {
-        Authorization: `Bearer ${localStorage.getItem("jwt")}`
+function requestConfig(){
+    return {
+        headers: {
+            Authorization: `Bearer ${localStorage.getItem("jwt")}`
+        }
     }
 }
 
 export function createAdminGame(game: Game){
-    return axios.post("/api/game/admin", game, requestConfig)
+    return axios.post("/api/game/admin", game, requestConfig())
 }
 
 export function createUserGame(game: Game){
-    return axios.post("/api/game/user", game, requestConfig)
+    return axios.post("/api/game/user", game, requestConfig())
 }
 
 export function fetchAllGames(){
-    return axios.get("/api/game", requestConfig)
+    return axios.get("/api/game", requestConfig())
         .then((response: AxiosResponse<Array<Game>>) => response.data)
 }
 
 export function getOneGame(gameId: string){
-    return axios.get("/api/game/" + gameId, requestConfig)
+    return axios.get("/api/game/" + gameId, requestConfig())
 }
 
 export function editGame(gameId: string){
-    return axios.put("/api/game/" + gameId, gameId, requestConfig)
+    return axios.put("/api/game/" + gameId, gameId, requestConfig())
```

Comment:
this function has a misleading name. 
you don't edit a game here. you are just changing the state if im right. 
also you are sending the gameID as path variable and as Requestbody. Maybe send the game in the Requestbody. 

```Javascript
export function editGame(game: Game) {
  return axios.put("api/game/" + game.id, game, requestConfig);
}
```
If you do this you must change the endpoints in your backend accordingly. But in this example your editGame() realy edits a game.

Reply:
oh yes i send the gameId twice 😄